lib/skate/detours/trip_modification.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Skate.Detours.TripModification do
+  @moduledoc """
+  A module that produces GTFS-flavored TripModification structs for a detour
+
+  More info: https://gtfs.org/realtime/reference/#message-tripmodifications
+  """
+
+  alias Schedule.Gtfs.RoutePattern
+  alias Schedule.Gtfs.Stop
+
+  defmodule Input do
+    @moduledoc """
+    Input for `Skate.Detours.TripModification.for/1`.
+    """
+
+    @type t :: %__MODULE__{
+            route_pattern: RoutePattern.t(),
+            missed_stops: [Stop.t()]
+          }
+
+    defstruct [:route_pattern, :missed_stops]
+  end
+
+  defmodule SelectedTrip do
+    @moduledoc """
+    Struct representing the `selected_trips` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-selectedtrips
+    """
+
+    @type t :: %__MODULE__{
+            trip_ids: [String.t()]
+          }
+
+    defstruct [:trip_ids]
+  end
+
+  defmodule Modification do
+    @moduledoc """
+    Struct representing the `modifications` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-modification
+    """
+
+    @type t :: %__MODULE__{
+            start_stop_selector: String.t(),
+            end_stop_selector: String.t()
+          }
+
+    defstruct [:start_stop_selector, :end_stop_selector]
+  end
```

Comment:
suggestion(non-blocking): if it's easy, I'd also love to get `last_modified_time` on this as well.

Reply:
[Here we go](https://github.com/mbta/skate/pull/2665/commits/a35857cb1975ada9a670bf22956e8ef5890cb9db)

I thought about just using `DateTime.utc_now`, but decided that having the time as an input was a better choice (especially because the `last_modified_time` might not actually be the time we create the `TripModification` 😅)